sharktank/sharktank/examples/paged_llm_v1.py

```diff
@@ -32,34 +34,73 @@ def __init__(
         self,
         model: PagedLlamaModelV1,
         tokenizer: InferenceTokenizer,
-        page_cache_size: int = 128,
         # Need to look at the model more for this.
         end_token: int = 2,
-        dump_bins: bool = False,
     ):
         self.model = model
         self.tokenizer = tokenizer
-        self.shared_cache_state = model.cache.allocate(page_cache_size)
-        self.free_pages = list(range(1, page_cache_size))
         self.end_token = end_token
-        self.dump_bins = dump_bins
 
     @property
     def block_seq_stride(self) -> int:
         return self.model.cache.block_seq_stride
 
-    def begin_batch(self, prompts: list[str]):
-        token_ids, seq_lens = self.tokenizer.encode(
-            prompts, pad_to_multiple_of=self.model.cache.pad_sequence_stride
-        )
-
-        token_ids = torch.tensor(token_ids, device=self.model.device)
-        seq_lens = torch.tensor(seq_lens, device=self.model.device)
-        if self.shared_cache_state is not None:
-            cache_state = self.shared_cache_state
+    def process_prompts(
```

Comment:
Could we change this to `preprocess_prompts`. Its closer what this step actually does.